### PR TITLE
[Docs] Updating benchmarkmarking.md with an updated sample output

### PR DESCRIPTION
**Pull request by jeffrade**
This PR is just a documentation update for someone (or myself) that looks into finishing up #7883 in the future.

Looked through #7883 and appears [ryanofsky's PR](https://github.com/bitcoin/bitcoin/pull/8873) setup the benchmarks, but there are `FIXME` comments to pull in data from `test/` to get a larger data set (assuming reason why 7883 is still open).

### CODE REVIEW

File: doc/benchmarking.md [hunk 1]
```diff
@@ -9,17 +9,31 @@ After compiling bitcoin-core, the benchmarks can be run with:
...
-Sleep100ms,10,0.205592155456543,0.210056066513062,0.104166316986084
-Trig,67108864,0.000000014997003,0.000000015448112,0.000000015188842
+# Benchmark, evals, iterations, total, min, max, median
+Base58CheckEncode, 5, 320000, 120.772, 7.49351e-05, 7.59374e-05, 7.54759e-05
+Base58Decode, 5, 800000, 122.833, 3.0467e-05, 3.11732e-05, 3.06304e-05
+Base58Encode, 5, 470000, 137.094, 5.81061e-05, 5.85109e-05, 5.84462e-05
+BenchLockedPool, 5, 530, 34.2023, 0.0128247, 0.0129613, 0.0129026
+CCheckQueueSpeedPrevectorJob, 5, 1400, 26.1762, 0.00365048, 0.00388629, 0.00367108
+CCoinsCaching, 5, 170000, 48.1074, 5.60229e-05, 5.72316e-05, 5.66214e-05
+CoinSelection, 5, 650, 34.6426, 0.0105801, 0.0107699, 0.010664
```
Comment (Bryantmeeder) [inline doc/benchmarking.md]: Mine

File: doc/benchmarking.md [hunk 2]
```diff
@@ -9,17 +9,31 @@ After compiling bitcoin-core, the benchmarks can be run with:
...
+DeserializeAndCheckBlockTest, 5, 160, 39.2084, 0.0483662, 0.0494199, 0.0490138
+DeserializeBlockTest, 5, 130, 23.8129, 0.0357731, 0.0373763, 0.0365858
+FastRandom_1bit, 5, 440000000, 38.1609, 1.72974e-08, 1.73882e-08, 1.73478e-08
+FastRandom_32bit, 5, 110000000, 72.8237, 1.29992e-07, 1.37014e-07, 1.30115e-07
+MempoolEviction, 5, 41000, 89.8883, 0.000432748, 0.000446857, 0.000438483
+PrevectorClear, 5, 5600, 47.9229, 0.00169952, 0.0017455, 0.00170315
+PrevectorDestructor, 5, 5700, 44.5498, 0.0015561, 0.00156977, 0.00156469
+RIPEMD160, 5, 440, 135.988, 0.0615496, 0.062268, 0.0617779
```
Comment (Bryantmeeder) [inline doc/benchmarking.md]: Bt

**Comment by MarcoFalke** [comment]
Could also mention that a help message can be printed by `-?`.

```
  -?
       Print this help message and exit

**Comment by jeffrade** [comment]
@MarcoFalke Added a help section for the `-?` option.  Small cleanup to match format of other md files in `doc/`.  [Direct link](https://github.com/jeffrade/bitcoin/blob/benchmark_output/doc/benchmarking.md) for visual.

**Comment by MarcoFalke** [comment]
ACK 7eaedea53214da7bcfc51b794b54de3663f02e03

**Comment by fanquake** [comment]
utACK b21244e

**Pull request merged**